plugin/route53/setup.go

```diff
@@ -75,11 +82,23 @@ func setup(c *caddy.Controller, f func(*credentials.Credentials) route53iface.Ro
 				if err != nil {
 					return c.Errf("invalid upstream: %v", err)
 				}
+			case "credentials_file":
+				args := c.RemainingArgs()
+				if len(args) > 0 {
+					sharedProvider.Profile = args[0]
+				}
+				if len(args) > 1 {
+					sharedProvider.Filename = args[1]
+				}
```

Comment:
```
if len(args) > 1 {
    sharedProvider.Filename = args[1]
```

That means additional args after 1 will be silently ignored?

Reply:
Yes, that's right. Would you like me to error out when len(args) > 2?

Reply:
I think this should be:

```
shared := &credentials.SharedCredentialsProvider{}
if c.NextArg() {
    shared.Profile = c.Val()
} else {
  return c.ArgErr()
}
if c.NextArg() {
    shared.Filename = c.Val()
    if c.NextArg() {
        return c.Err("number of arguments is too damn high")
    }
}
providers = append(providers,  shared)
```

Something like that. You don't check for required `PROFILE` argument currently. Also can you add a test for this.

Reply:
Updated this one. Just realized that there's no need for that extra NextArg that checks extra args. This is something magically handled by Caddy Dispenser itself, however, still added another test case to validate that it works as expected.